BlogTemplate/Pages/Edit.cshtml.cs

```diff
@@ -35,8 +35,8 @@ public void OnGet()
 
         private void InitializePost()
         {
-            string slug = RouteData.Values["slug"].ToString();
-            newPost = oldPost = _dataStore.GetPost(slug);
+            int id = Convert.ToInt32(RouteData.Values["id"]);

```

Comment:
It looks like only OnGet uses this method, can we just move the code up there?  Also, we could change the RouteData.Values["id"] to be a parameter for Get like this:

        public void OnGet([FromRoute] int id)